=== modified file 'src/user-list.vala'
--- a/src/user-list.vala
+++ b/src/user-list.vala
@@ -20,7 +20,7 @@
 
 int remote_server_field_sort_function (RemoteServerField? item1, RemoteServerField? item2)
 {
-    string[] sorted_fields = { "domain", "username", "email", "password" };
+    string[] sorted_fields = { "domain", "x2gosession" , "username", "email", "password"};
     foreach (var field in sorted_fields)
     {
         if (item1.type == field)
@@ -702,6 +702,13 @@
                         entry.text = default_value;
                         widget = entry;
                     }
+                    else if (field.type == "x2gosession")
+                    {
+                        var prompt = add_prompt (_("X2Go Session:"));
+                        prompt.text = default_value;
+                        prompt.sensitive = true;
+                        widget = prompt;
+                    }
                     else if (field.type == "domain")
                     {
                         string[] domainsArray = {};
@@ -1077,6 +1084,12 @@
                 var answer = field != null ? field.text : "";
                 UnityGreeter.singleton.respond (answer);
             }
+            else if (text == "x2gosession:")
+            {
+                Gtk.Entry field = current_remote_fields.get ("x2gosession") as Gtk.Entry;
+                var answer = field != null ? field.text : "";
+                UnityGreeter.singleton.respond (answer);
+            }
         }
         else
             base.show_prompt_cb (text, type);
